docs/contributing/coding-guidelines/ros-nodes/message-guidelines.md

```diff
@@ -0,0 +1,151 @@
+# Message guidelines
+
+## Format
+
+All messages should follow [ROS message description specification](https://docs.ros.org/en/galactic/Concepts/About-ROS-Interfaces.html#background).
+
+The accepted formats are:
+
+- `.msg`
+- `.srv`
+- `.action`
+
+## Default units
+
+All the fields by default have the following units depending on their types:
+
+| type           | default unit  |
+| -------------- | ------------- |
+| distance       | meter (m)     |
+| angle          | radians (rad) |
+| time           | second (s)    |
+| speed          | m/s           |
+| velocity       | m/s           |
+| acceleration   | m/s²          |
+| angular vel.   | rad/s         |
+| angular accel. | rad/s²        |
+
+!!! warning ""
+
+    If a field in a message has any of these default units, don't add any suffix or prefix denoting the type.
+
+## Non-default units
+
+For non-default units, use following suffixes:
+
+| type     | non-default unit | suffix  |
+| -------- | ---------------- | ------- |
+| distance | nanometer        | `_nm`   |
+| distance | micrometer       | `_um`   |
+| distance | millimeter       | `_mm`   |
+| distance | kilometer        | `_km`   |
+| angle    | degree (deg)     | `_deg`  |
+| time     | nanosecond       | `_ns`   |
+| time     | microsecond      | `_us`   |
+| time     | millisecond      | `_ms`   |
+| time     | minute           | `_min`  |
+| time     | hour (h)         | `_hour` |
+| velocity | km/h             | `_kmph` |
+
+!!! tip ""
+
+    If a unit that you'd like to use doesn't exist here, [create an issue/PR](https://github.com/autowarefoundation/autoware-documentation/issues) to add it to this list.
+
+## Message field types
+
+For list of types supported by the ROS interfaces [see here](https://docs.ros.org/en/galactic/Concepts/About-ROS-Interfaces.html#field-types).
+
+Also copied here for convenience:
+
+| Message Field Type | C++ equivalent   |
+| ------------------ | ---------------- |
+| `bool`             | `bool`           |
+| `byte`             | `uint8_t`        |
+| `char`             | `char`           |
+| `float32`          | `float`          |
+| `float64`          | `double`         |
+| `int8`             | `int8_t`         |
+| `uint8`            | `uint8_t`        |
+| `int16`            | `int16_t`        |
+| `uint16`           | `uint16_t`       |
+| `int32`            | `int32_t`        |
+| `uint32`           | `uint32_t`       |
+| `int64`            | `int64_t`        |
+| `uint64`           | `uint64_t`       |
+| `string`           | `std::string`    |
+| `wstring`          | `std::u16string` |
+
+### Arrays
+
+For arrays, use `unbounded dynamic array` type.
+
+Example:
+
+```text
+int32[] unbounded_integer_array
+```
+
```

Comment:
@maxime-clem could you also check here following from https://github.com/autowarefoundation/autoware_msgs/pull/11#discussion_r854781463

Reply:
Looks good to me, thank you :+1: